doc/plutus-core-spec/cardano/builtins5.tex

```diff
@@ -0,0 +1,252 @@
+$b_i \in \{\bitzero,\bitone\}$ (and $n$ necessarily a multiple of 8).  Strictly
+we should use the functions $\bitsof$ and $\bytesof$ of
+Section~\ref{sec:notation-bytestrings} to convert back and forth between
+bytestrings and bitstrings, but we elide this for conciseness and reduce
+ambiguity by using lower-case names like $b$,$c$ and $d$ for bits and upper-case
+names like $B$ for bytes. We denote the complement of a bit
+$b \in \{\bitzero,\bitone\}$ by $\bar{b}$, so $\bar{\bitzero} = \bitone$ and
+$\bar{\bitone} = \bitzero$.
+
+
+\setlength{\LTleft}{-6mm}  % Shift the table left a bit to centre it on the page
+\begin{longtable}[H]{|l|p{50mm}|p{40mm}|c|c|}
+    \hline
+    \text{Function} & \text{Signature} & \text{Denotation} & \text{Can} & \text{Note} \\
+    & & & fail? & \\
+    \hline
+    \endfirsthead
+    \hline
+    \text{Function} & \text{Type} & \text{Denotation} & \text{Can} & \text{Note}\\
+    & & & fail? & \\
+    \hline
+    \endhead
+    \hline
+    \caption{Built-in functions, batch 5}
+    \endfoot
+    \caption[]{Built-in functions, batch 5}
+    \label{table:built-in-functions-5}
+    \endlastfoot
+    \TT{andByteString} & $[\ty{bool}, \ty{bytestring}, \ty{bytestring}] $ \text{$\;\; \to \ty{bytestring}$} & $\Xand$ & No & \ref{note:bitwise-logical-ops} \\[2mm]
+    \TT{orByteString} & $[\ty{bool}, \ty{bytestring}, \ty{bytestring}] $ \text{$\;\; \to \ty{bytestring}$} & $\Xor$ & No  & \ref{note:bitwise-logical-ops} \\[2mm]
+    \TT{xorByteString} & $[\ty{bool}, \ty{bytestring}, \ty{bytestring}] $ \text{$\;\; \to \ty{bytestring}$} & $\Xxor$ & No & \ref{note:bitwise-logical-ops} \\[2mm]
+    \TT{complementByteString} & $[\ty{bytestring}] \to \ty{bytestring}$
+                              &  $ b_{n-1}{\cdots}b_0 \mapsto \bar{b}_{n-1}{\cdots}\bar{b}_0$  & No & \\[2mm]
+    \TT{shiftByteString} & $[\ty{bytestring}, \ty{integer}] $ \text{$\;\; \to \ty{bytestring}$} &  $\mathsf{shift}$ & No & \ref{note:shift} \\[2mm]
+    \TT{rotateByteString} & $[\ty{bytestring}, \ty{integer}] $ \text{$\;\; \to \ty{bytestring}$} &  $\mathsf{rotate}$ & No & \ref{note:rotate}\\[2mm]
+    \TT{countSetBits} & $[\ty{bytestring}] \to \ty{integer}$
+    & $b_{n-1}{\cdots}b_0 $ \text{$\mapsto \left|\{i: b_i =1\}\right|$} & No & \\[2mm]
+    \TT{findFirstSetBit} & $[\ty{bytestring}] \to \ty{integer}$ & $\mathsf{ffs}$ & No & \ref{note:ffs}\\[2mm]
+    \TT{readBit} & $[\ty{bytestring}, \ty{integer}] $ \text{$\;\; \to \ty{bytestring}$}
+                                        & $(b_{n-1}{\cdots}b_0, i) $ \text{$\mapsto \begin{cases}
+                                        b_i & \text{if $0 \leq i \leq n-1$}\\
+                                        \errorX & \text{otherwise}
+                                        \end{cases}$}
+                                        & Yes & \\[14mm]  %% Odd, but it gives reasonable spacing
+    \TT{writeBits} & $[\ty{bytestring}, \listOf{\ty{integer}}, $
```

Comment:
This is outdated, and I don't think `writeBits` will make it to Chang+1, as node 10.0 release is imminent, so you may want to remove this.

Reply:
See the [previous comment](https://github.com/IntersectMBO/plutus/pull/6426#issuecomment-2400443074).